test/collection/map_mutable_extensions_test.dart

```diff
@@ -207,26 +198,22 @@ void testMutableMap(KtMutableMap<K, V> Function<K, V>() emptyMap,
 
       expect(i.hasNext(), isFalse);
     });
+  });
 
-    group("remove doesn't work", () {
-      test("iterator is iterates", () {
-        final pokemon = mutableMapFrom({
-          1: "Bulbasaur",
-          2: "Ivysaur",
-        });
-        final KtMutableIterator<KtMapEntry<int, String>> i = pokemon.iterator();
-        expect(i.hasNext(), isTrue);
-        final next = i.next();
-        expect(next.key, 1);
-        expect(next.value, "Bulbasaur");
-
-        // TODO replace error assertion with value assertion when https://github.com/passsy/dart_kollection/issues/5 has been fixed
-        final e = catchException(() => i.remove());
-        expect(e, equals(const TypeMatcher<UnimplementedError>()));
-        // removed first item
-        //expect(pokemon, mapFrom({2: "Ivysaur"}));
-      });
+  test("remove item from map via iterator", () {
+    final pokemon = mutableMapFrom({
+      1: "Bulbasaur",
+      2: "Ivysaur",
     });
+    final KtMutableIterator<KtMapEntry<int, String>> i = pokemon.iterator();
+    expect(i.hasNext(), isTrue);
+    final next = i.next();
+    expect(next.key, 1);
+    expect(next.value, "Bulbasaur");
+
+    i.remove();
+    // removed first item
+    expect(pokemon, mapFrom({2: "Ivysaur"}));
   });
```

Comment:
This failing test needs to pass, before `remove()` works as intended

Reply:
Should pass now